File: python-version/LC1399/count_largest_group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        mp = dict()
        max_size, ans = 0, 0
        for i in range(1, n+1):
            sm = self.__get_digit_sum(i)
            cur_list = mp.get(sm, [])
            cur_list.append(i)
            mp[sm] = cur_list
            if len(cur_list) > max_size:
                max_size = len(cur_list)
                ans = 1
            elif len(cur_list) == max_size:
                ans += 1
        return ans

    @staticmethod
    def __get_digit_sum(x: int) -> int:
        ans, t = 0, x
        while t > 0:
            ans += t % 10
            t //= 10
        return ans

File: go-version/leetcode/LC1399/count_largest_group.go
package LC1399

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/4/23 下午10:01
 * @FilePath: count_largest_group
 * @Description: 遍历分组计数
 */

func countLargestGroup(n int) int {
	mp := make(map[int][]int)
	ans, maxSize := 0, 0
	for i := 1; i <= n; i++ {
		sum := getDigitSum(i)
		mp[sum] = append(mp[sum], i)
		if len(mp[sum]) > maxSize {
			maxSize = len(mp[sum])
			ans = 1
		} else if len(mp[sum]) == maxSize {
			ans++
		}
	}
	return ans
}

func getDigitSum(x int) int {
	ans, t := 0, x
	for t > 0 {
		ans += t % 10
		t /= 10
	}
	return ans
}
